sre_maturity.adoc

```diff
@@ -0,0 +1,164 @@
+Communication is important in your team and you hire diverse people based on soft skills & leadership rather than specific technologies.
+You may also have folks from previous "disciplines". For example, sys admins, or experienced developers. Or, even some folks who have experience in "devops". They can bring great skills, experience and value.
+You are likely to see resistance to change when making this transition with existing team members, so retraining (or even unlearning) can help. However, there may be inevitable difficult personnel decisions to be made.
+At this phase, having buy-in from executive level leadership is essential.
+Without this it will be very difficult to get their support later, and eventually their engagement.
+
+The product manager for you service should have a good understanding of SRE.
+They should help you advocate for service features that make the service more reliable.
+
+=== Phase 2: The 'Walk' Phase
+
+* Starting to get Exec support (improvement on 'buy-in')
+
+=== Phase 3: The 'Run' Phase
+
+* Exec engagement
+
+== Observability
+
+=== Phase 1: The ‘Crawl’ Phase
+
+The foundations for monitoring and observing your service are in place.
+Being able to measure something meaningful about your service is important, and supplements any feedback you get from initial users.
+A common way to do this is via metrics exposed by your service.
+
+In this phase with a well established team that understands common SRE practices, SLOs would start to appear for their service(s) as a way of placing measuring sticks around the effectiveness of scaling efforts. Error budgets wouldn't be tracked, at least not formally.
+
+The SLO might only be one around service availability and it may not even be formally tracked at the start of this phase (as one is first learning to walk). Towards the end of the phase, as the team transitions to "running," the they may start to wonder how to enforce that availability SLO and other new ones they want to put into place. 
+
+There's a graduation process with SLOs from "informing" to "enforcing". You can use SLOs to inform decisions in this early stage. As they get refined and wider awareness & support, you can think about enforcement of an error budget policy.
+
+
+=== Phase 2: The 'Walk' Phase
+
+You have established patterns and guidelines for observability signals such as metrics, logs, errors etc...
+You are using these patterns to speed up debugging of issues and correlating between signals easier, thereby reducing your Mean Time To Recovery (MTTR).
+Your alerts have matured and rarely give false positives as a result of having these patterns and refining queries.
+
+* SLOs - SLOs have been iterated on, and you have more than 1 meaningful SLO.
+* You understand and track error budgets
+* Actionable Alerts
+* Self healing
+  * Avoid SRE having sub-systems that automagically work around the inefficiencies of upstream - needs a good argument why it can’t be fixed upstream - pushing left
```

Comment:
As someone from outside of your team, I don't understand what this means.  Can you expand a little, or give a clarifying example?

Reply:
Ah -- I've seen plenty of these types of gadgets paper over things that should be solved in the upstream codebase.  So I agree with this call-out because this does two important negative things:

1) SRE teams not getting proper attention from development teams for issues affecting the service
2) SRE teams owning an ever increasing amount of code
2) Any amount of divergence from upstream becomes a support/docs burden. Nuances are only known about in small pockets.